runner/app/live/trickle/media.py

```diff
@@ -97,6 +98,8 @@ def decode_runner():
                     logging.exception(f"Error in decode_av, retrying {retry_count}/{MAX_DECODER_RETRIES}", stack_info=True)
                 else:
                     logging.exception("Error in decode_av, maximum retries reached", stack_info=True)
+                    # force write end of pipe to close to terminate trickle subscriber
+                    write_fd.close()
```

Comment:
Maybe move this to outside the loop, like close to `os.close(in_pipe)` in case we exit through some other path? There's a `break` on L89 👀 

Or even better if all that clean up would be in a finally block, but we'd need to indent the whole `while` loop here 😫

Reply:
I didn't really want to change the behavior too much since IIRC we'd get _another_ asyncio exception if trying to close an already-closed FD which would be the typical behavior under most normal shutdowns. So we'd need nested exception handling in addition to the `finally`, which is also annoying.

I agree it is kind of fragile if another exit path slips through in the future. Moving closer to the `in_pipe` is an option.

> There's a break on L89 👀

Good spot, the `write_fd` would already be closed in this case. That just prevents us from retrying on an input that we already know to be closed.

Reply:
sounds good let's keep this minimal to include it in the deploy tmr